From 3eb0a40f494fa3867fe8a68c590be5bfda09b485 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Thu, 08 Jul 2021 10:16:30 +0300
Subject: [PATCH] [SW-15543] greco: secure: remove gpio assert check

In case of greco, when SW_ALL reset is done PSOC
block does not get reset. In few platforms this
asserts/deasserts host gpio used for i2c slave
debug mode. Due to this next time preboot boots
it goes in idle state without even completing
preboot command loop. This is "revert me" patch
added until this issue is resolved.

Change-Id: Ib06109668f035bec3ba95d01cfb6f5e1f34d173d
---

diff --git a/arch/greco_arc_gnu/common.c b/arch/greco_arc_gnu/common.c
index fc1f04b..68e2e36 100644
--- a/arch/greco_arc_gnu/common.c
+++ b/arch/greco_arc_gnu/common.c
@@ -57,24 +57,6 @@
 #define mmPCIE_WRAP_VUART_TX_STATUS     0xC15F64
 #define HOST_GPIO_I2C_DEBUG	44
 
-/* HOST_GPIO is set through mmPSOC_GPIO2, each mmPSOC_GPIO serve 16 GPIOs.*/
-bool preboot_host_gpio_asserted(void)
-{
-	u32 reg, mask;
-
-	mask = BIT(HOST_GPIO_I2C_DEBUG % 16);
-
-	/* configure as input */
-	regmap_writel(mask, mmPSOC_GPIO2_OUTENCLR);
-
-	/* clear ALT function for gpio */
-	regmap_writel(mask, mmPSOC_GPIO2_ALTFUNCCLR);
-
-	reg  = regmap_readl(mmPSOC_GPIO2_DATA);
-
-	return reg & mask;
-}
-
 u64 lkd_dev_spec_get_img_addr(uintptr_t base)
 {
 	/* Return full address */
